todo_app/app.py

```diff
@@ -17,41 +21,109 @@ def del_DB(db):
 def create_app(db = None):
     app = Flask(__name__)
     app.config.from_object('todo_app.flask_config.Config')
+    app.config['LOGIN_DISABLED'] = os.getenv('LOGIN_DISABLED')
 
     if db == None:
         db = get_DB()
 
-    @app.route('/', methods=["GET"])
+    #login manager
+    login_manager = LoginManager()
+
+    client = WebApplicationClient(os.getenv('GITHUB_CLIENTID'))
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        authorization_endpoint = 'https://github.com/login/oauth/authorize'
+        request_uri = client.prepare_request_uri(
+            authorization_endpoint
+        )
+        return redirect(request_uri)
+
+    @login_manager.user_loader
+    def load_user(user_id):
+        if user_id in users:
+            return users[user_id] 
+        else:
+            return None
+
+
+    login_manager.init_app(app)
+
+    @app.route('/users/callback', methods=["GET"])
+    def callback_func():
+        token_endpoint='https://github.com/login/oauth/access_token'
+        token_url, headers, body = client.prepare_token_request(
+            token_endpoint,
+            authorization_response=request.url,
+            redirect_url=request.base_url,
+            client_secret=os.getenv('GITHUB_CLIENTSECRET')
+        )
+        token_response = requests.post(
+            token_url,
+            headers=headers,
+            data=body,
+            auth=(os.getenv('GITHUB_CLIENTID'), os.getenv('GITHUB_CLIENTSECRET'))
+        )
+        # Parse the tokens!
+        client.parse_request_body_response(token_response.content.decode())
+
+        # get user details
+        userinfo_endpoint = 'https://api.github.com/user'
+        uri, headers, body = client.add_token(userinfo_endpoint)
+        userinfo_response = requests.get(uri, headers=headers, data=body)
+        if userinfo_response.json().get("login"):
+            login = userinfo_response.json()["login"]
+            unique_id = userinfo_response.json()["id"]
+            email = userinfo_response.json()["email"]
+            avatar = userinfo_response.json()["avatar_url"]
+        else:
+            return "User email not available or not verified by Github.", 400
+        user = User(unique_id,login, email, avatar)
+        if login_user(user):
+            users[str(unique_id)] = user
+            return redirect(url_for("index_get"))
```

Comment:
Worth considering here what you want to do if this fails - otherwise why have the `if` statement?